NewsApp/NewsApp/Controller/ProfileController.swift

```diff
@@ -80,13 +87,23 @@ extension ProfileController: UIPickerViewDataSource {
 }
 
 extension ProfileController: UIPickerViewDelegate {
-    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
-        return Country.allCases[row].name
+//    Pour utiliser sans FlagKit, décommenter cette fonction et supprimer func pickerView(viewForRow)
+
+//    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+//        return Country.allCases[row].name
+//    }
+
+    func pickerView(_ pickerView: UIPickerView, viewForRow row: Int, forComponent component: Int, reusing view: UIView?) -> UIView {
+        let flagCountryImage = Flag(countryCode: Country.allCases[row].name.uppercased())?.originalImage
+        return UIImageView(image: flagCountryImage)
```

Comment:
Comme je vois 3 occurrences de Flat sur le fichier, je pense qu'il aurait été plus simple d'ajouter un service qui retourne une image en fonction d'un country

Pour te donner une une idée

protocol FlagService {
     func imageFor(country: Country?) -> Image?
}

Comme ça seul ton service connait FlagKit, et ton controller ne connait que FlagService, si tu veux changer l'implémentation de FlagKit vers autre chose, tu changes ce qu'il y a à l'intérieur de ton service.

Reply:
fix in "correction PR"